src/ui/components/EntryContextMenu/constants.ts

```diff
@@ -155,17 +156,28 @@ export const getEntryContextMenu = (): ContextMenuItem[] => [
     },
     {
         ...CONTEXT_MENU_COPY,
-        scopes: [EntryScope.Dataset],
+        scopes: [EntryScope.Connection],
         permissions: {admin: true, edit: true, read: false, execute: false},
         isStrictPermissions: true, // strict check with disallow when there are no permissions object
         isVisible(args) {
-            if (args.entry?.workbookId) {
+            const entry = args.entry;
+            const isFileConnection =
+                entry?.scope === EntryScope.Connection &&
+                (entry?.type === ConnectorType.File || entry?.type === ConnectorType.GsheetsV2);
```

Comment:
I think we should reuse [this](https://github.com/datalens-tech/datalens-ui/blob/main/src/ui/units/connections/constants/connectors.ts#L4) array. Maybe it should be located in `ui/constants/connections.ts`. And u can rename it to `isS3BasedConnector`